src/backend/distributed/shardsplit/shardsplit_decoder.c

```diff
@@ -62,6 +80,192 @@ _PG_output_plugin_init(OutputPluginCallbacks *cb)
 	/* actual pgoutput callback will be called with the appropriate destination shard */
 	pgoutputChangeCB = cb->change_cb;
```

Comment:
optional: could rename it to originalChangeDB, since the only pgoutput-specific logic appears to be the load_external_function argument

Reply:
OK